include/mata/alphabet.hh

```
@@ -169,10 +169,151 @@ namespace Mata {
+            return m_symbols.difference(symbols);
+        }
+
+        std::string reverse_translate_symbol(const Symbol symbol) const override {
+            if (m_symbols.find(symbol) == m_symbols.end()) {
+                throw std::runtime_error("Symbol '" + std::to_string(symbol) + "' is out of range of enumeration.");
+            }
+            return std::to_string(symbol);
+        }
+
+    private:
+        EnumAlphabet& operator=(const EnumAlphabet& rhs);
+
+    public:
+        /**
+         * @brief Expand alphabet by symbols from the passed @p symbol_names.
+         *
+         * Adding a symbol name which already exists will throw an exception.
+         * @param[in] symbols Vector of symbols to add.
+         */
+        void add_symbols_from(const Mata::Util::OrdVector<Symbol>& symbols) { m_symbols.insert(symbols); }
+
+        /**
+         * @brief Expand alphabet by symbols from the passed @p alphabet.
+         *
+         * @param[in] symbols_to_add Vector of symbols to add.
+         */
+        void add_symbols_from(const EnumAlphabet& alphabet) { m_symbols.insert(alphabet.get_alphabet_symbols()); }
+
+        EnumAlphabet(std::initializer_list<Symbol> symbols) : EnumAlphabet(symbols.begin(), symbols.end()) {}
+
+        template <class InputIt>
+        EnumAlphabet(InputIt first, InputIt last) : EnumAlphabet() {
+            for (; first != last; ++first) {
+                add_new_symbol(*first);
+            }
+        }
+
+        EnumAlphabet(std::initializer_list<std::string> l) : EnumAlphabet(l.begin(), l.end()) {}
+
+        Symbol translate_symb(const std::string& str) override {
+            Symbol symbol;
+            std::istringstream stream{ str };
+            stream >> symbol;
+            return symbol;
+
+            // TODO: How can the user specify to throw exceptions when we encounter an unknown symbol? How to specify that
+            //  the alphabet should have only the previously fixed symbols?
```

I think we should just throw an exception if we are trying to translate "unknown" symbol.

---

I agree. I will modify it and if no errors arise, I will commit the changes.

---

Modified.